lib/stages/basebandReadout.cpp

```diff
@@ -47,7 +48,7 @@ basebandReadout::basebandReadout(Config& config, const string& unique_name,
     oldest_frame(-1),
     frame_locks(_num_frames_buffer),
     // Over allocate so we can align the memory.
-    baseband_data(std::make_unique<uint8_t[]>(_num_elements * _max_dump_samples + 15)),
+    data_buffer(_num_elements * _max_dump_samples + 15),
```

Comment:
I think this should be `2 * (_num_elements * _max_dump_samples + 15)`, since even if there are no writes queued you can't guarantee that more than length/2 of contiguous memory is available.